x-pack/plugin/core/src/main/java/org/elasticsearch/license/LicenseUtils.java

```diff
@@ -63,27 +62,23 @@ public static boolean licenseNeedsExtended(License license) {
      * Checks if the signature of a self generated license with older version needs to be
      * recreated with the new key
      */
-    public static boolean signatureNeedsUpdate(License license, DiscoveryNodes currentNodes) {
+    public static boolean signatureNeedsUpdate(License license, ClusterState currentState) {
         assert License.VERSION_ENTERPRISE == License.VERSION_CURRENT : "update this method when adding a new version";
 
         String typeName = license.type();
         return (LicenseType.isBasic(typeName) || LicenseType.isTrial(typeName)) &&
         // only upgrade signature when all nodes are ready to deserialize the new signature
             (license.version() < License.VERSION_CRYPTO_ALGORITHMS
-                && compatibleLicenseVersion(currentNodes) >= License.VERSION_CRYPTO_ALGORITHMS);
+                && compatibleLicenseVersion(currentState) >= License.VERSION_CRYPTO_ALGORITHMS);
     }
 
-    public static int compatibleLicenseVersion(DiscoveryNodes currentNodes) {
-        return getMaxLicenseVersion(currentNodes.getMinNodeVersion());
+    public static int compatibleLicenseVersion(ClusterState state) {
```

Comment:
Could this method be removed and callers switched to getMaxLicenseVersion since the cluster state is never consulted?

Reply:
Yep, I also cleaned up some related plumbing that only existed to get the `ClusterState` here, and tweaked the name of this method and added Javadoc. Good catch, thanks.